rotkehlchen/exchanges/kraken.py

```diff
@@ -782,3 +779,282 @@ def query_online_income_loss_expense(
+
+            exchange_uuid = (
+                str(base_trade.event_identifier) +
+                str(timestamp)
+            )
+            if fee_part is not None and kfee_part is not None:
+                fee = Fee(fee_part.amount.balance.amount)
+                fee_asset = fee_part.amount.asset
+                # Here we should create a ledger action for kfee
+            elif (None, None) == (fee_part, kfee_part):
+                fee = Fee(ZERO)
+                fee_asset = received_asset
+            elif fee_part is not None:
+                fee = Fee(fee_part.amount.balance.amount)
+                fee_asset = fee_part.amount.asset
+            elif kfee_part is not None:
+                fee = Fee(kfee_part.amount.balance.amount)
+                fee_asset = A_KFEE
+
+            trade = Trade(
+                timestamp=timestamp,
+                location=Location.KRAKEN,
+                base_asset=base_asset,
+                quote_asset=quote_asset,
+                trade_type=trade_type,
+                amount=AssetAmount(amount),
+                rate=rate,
+                fee=fee,
+                fee_currency=fee_asset,
+                link=exchange_uuid,
+            )
+            trades.append(trade)
+        return trades
+
+    def query_kraken_ledger_actions(self) -> None:
+        with self.ledger_lock:
+            range_query_name = f'{self.location}_history_events_{self.name}'
+            last_query = self.db.get_used_query_range(range_query_name)
+            now = ts_now()
+            from_ts = Timestamp(0)
+            if last_query is not None:
+                last_query_ts = last_query[1]
+                from_ts = Timestamp(last_query_ts + 1)
+
+            response, _ = self.query_until_finished(
+                endpoint='Ledgers',
+                keyname='ledger',
+                start_ts=from_ts,
+                end_ts=now,
+                extra_dict={},
+            )
+            # Group related events
+            raw_events_groupped = defaultdict(list)
+            for raw_event in response:
+                raw_events_groupped[raw_event['refid']].append(raw_event)
+
+            new_events = []
+            for events in raw_events_groupped.values():
+                try:
+                    events = sorted(
+                        events,
+                        key=lambda x: deserialize_timestamp_from_kraken(x['time']),
+                    )
+                except DeserializationError as e:
+                    self.msg_aggregator.add_error(
+                        f'Failed to read timestamp in kraken event group '
+                        f'{events} due to {str(e)}',
+                    )
+                    continue
+                group_events = []
+                for idx, raw_event in enumerate(events):
+                    found_unknown_event = False
+                    try:
+                        timestamp = deserialize_timestamp_from_kraken(raw_event['time'])
```

Comment:
This function is huge. Everything inside this try statement should be on its own function.

So something like `historic_event_from_kraken()`.